### ISSUE TITLE
Different Animations/Look for Right Mouse Button and Middle Mouse Button on Click and Drag

### ISSUE THREAD

**Issue by KenzKD** [mulaRahul/keyviz#272]
I use 3d Software and have to frequently use the Right Mouse Button and Middle Mouse Button. Currently there is no way to differentiate the Mouse Clicks unless you enable the show mouse events.

**Describe the solution you'd like**
On Right click perhaps we could change the shape to like a square and for the middle mouse click we could do to a triangle perhaps or some other shapes.

**Describe alternatives you've considered**
Instead of shapes we could also change the color on right click and middle mouse click. I think this is should be can be an additional feature instead of the main solution as you can have issue with color blindness and the above Solution would be more effective for that.

**Comment by Several-Record7234** [comment]
I second this one! 😄

I have a simpler (and hopefully more visually-analogous) suggestion for differentiating between mouse buttons: because the buttons are normally arranged left to right like: [ ] 0 [ ] then having an outlined set of three buttons appear for each click, with the triggered one highlighted/filled-in, would give visual context to whether the left, middle, or right mouse button has been clicked. This could then be presented in the same elegant visual style as the rest of the keystrokes, and would need no legend to decode the relationship between the on-screen indication and the physical mouse buttons.

**Comment by mulaRahul** [comment]
I think going with different shapes or icons beside the cursor will be great. The same will be an optional addition to the current one. I would like more inputs about the shapes, here's something I came up with -

![image](https://github.com/user-attachments/assets/6308d753-90c5-4827-a604-74c8ee096aa4)

**Comment by KenzKD** [comment]
> ![image](https://private-user-images.githubusercontent.com/96373135/367673309-6308d753-90c5-4827-a604-74c8ee096aa4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY0NzE2NzUsIm5iZiI6MTcyNjQ3MTM3NSwicGF0aCI6Ii85NjM3MzEzNS8zNjc2NzMzMDktNjMwOGQ3NTMtOTBjNS00ODI3LWE2MDQtNzRjOGVlMDk2YWE0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE2VDA3MjI1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIxOTY2MjJhNzE4OTNjMDdjNTcwZTIzZTgwZTgwNjkyNzA0ZmIwMDhmNjRkYzViOGZkMzg0N2VhMjRlM2ZmOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UbvUrc9tLFgIzPQbQGVBn6zeAU4Yxn2Ws1QGpM-59A8)

This looks Amazing. I wonder how the Animations would look tho since I'm not sure how animating from the bottom corners would look when the general mouse highlight is already surrounding the mouse. Im sure that can be solved after playing around with the animation a bit. The graphics look Solid tho

**Comment by Several-Record7234** [comment]
I like the idea of the shaped animations to denote which button was pressed, but am concerned that the visual distinction is too subtle between them, especially when a button is clicked momentarily and not held down. I make tutorial videos and clarity trumps aesthetics every time - if it looks nice but is hard to parse or catch the information then it's second place to something that is not as polished but reliably conveys that vital information.

So, I would tentatively suggest that the middle button's oval cursor is fine (it looks like a mousewheel, after all!), but to make the right button more distinct as a 'D' shape, and the left button being the mirror image of that. You might even go as far as a right-facing triangle for the right button (and vice versa for the left button), but I leave that to you to decide where you want to draw the line between design elegance and reliable interpretation by the watcher.

**Comment by mulaRahul** [comment]
That's a great viewpoint from the viewers perspective, and yes I think the arrows will be much clear.

![image](https://github.com/user-attachments/assets/861438a7-c3a0-4446-9c6a-a8929d0fee5c)

That's the feedback I'm looking for, thanks 👍!

**Comment by KenzKD** [comment]
> ![image](https://private-user-images.githubusercontent.com/96373135/367683423-861438a7-c3a0-4446-9c6a-a8929d0fee5c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY0NzQyMjIsIm5iZiI6MTcyNjQ3MzkyMiwicGF0aCI6Ii85NjM3MzEzNS8zNjc2ODM0MjMtODYxNDM4YTctYzNhMC00NDQ2LTljNmEtYTg5MjlkMGZlZTVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTE2VDA4MDUyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE4ODI1YjEyN2Q1OTU5YzA0YjZhMjYyNWNmMzY4OGJkODE4YTc3YWExYjI4ODA0ODRkYjNlN2Y4NzViMThjYjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.llDCsf37EDvq6RMr5ihXFRgX0vV8YzeFJPUBpOkgC2Y)

With this change you can also change the animation anchor to the middle as well. So you will essentially be blending the shapes. You can also try adding a parameter to the shapes to change the curves. Some ppl might want it sharper and others might want it curvier.

**Comment by yw4z** [comment]
![Screenshot-20240921112142](https://github.com/user-attachments/assets/b82ee5b8-a671-4667-a78a-0ca68ba315c5)

here is 2 alternative from me

**Comment by GitHubGoody** [comment]
I like the second of @yw4z's options.

**Comment by dimateos** [comment]
Is the middle mouse properly detected in non windows-systems? https://github.com/mulaRahul/keyviz/issues/346 would have to be addressed I guess.
